### PR TITLE
SearchViewListener: filter collection

### PR DESCRIPTION
**Pull request by ADmad**
What this PR does:
- Adds ability to specify the filter collection to use.
- ~~Adds `beforePaginate` callback which automatically applies `search` finder to the query.~~

Referring to changes #171:

> scaffold.index_filters = null|false: default, disable filters

I don't think not showing any filters is a good default. If you don't want any filter inputs to be generated one should just not load the listener or disable it.

> scaffold.index_filters = true: auto-import default search collection
> scaffold.index_filters = 'crud': auto-import a specific search collection

The `collection` option I added to the listener takes care of this.

> scaffold.index_filters = []: specific array of filter settings

The listener already has support for `fields` option to achieve the same. I have documented it.

### CODE REVIEW

**Comment by josegonzalez** [comment]
Needs docs before a review :)

**Comment by ADmad** [comment]
There isn't even a page yet for this listener :disappointed:

**Comment by ADmad** [comment]
Hmm on second though not sure if adding `beforePaginate()` to this listener  to apply search finder is necessary. Search plugin's search listener already does that, I should just add `collection` config to it too.

**Comment by ADmad** [comment]
So apparently there was an empty docs page for filters :)

File: docs/_partials/pages/index/filters.rst [hunk 1]
```diff
@@ -1,2 +1,35 @@
...
+.. code-block:: php
+
+    <?php
+    class SamplesController extends AppController {
+
+        public function index() {
+            $this->Crud->addListener('Crud.Crud');
+            $this->Crud->addListener('Crud.ViewSearch', [
+                // Indicates whether is listener is enabled.
+                'enable' => true,
```
Comment (josegonzalez) [inline docs/_partials/pages/index/filters.rst]: Shouldn't it be enabled?

File: docs/_partials/pages/index/filters.rst [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+
+                // Whether to use auto complete for select fields. Default `true`.
+                'autocomplete' => true,
```
Comment (josegonzalez) [inline docs/_partials/pages/index/filters.rst]: Does this require extra setup?

**Comment by ADmad** [comment]
Docs fixed

File: src/Listener/ViewSearchListener.php [hunk 3]
```diff
@@ -106,6 +117,10 @@ protected function _deriveFields()
                 'type' => 'text'
             ];
 
+            if (substr($field, -3) === '_id' && $field !== '_id') {
```
Comment (josegonzalez) [inline src/Listener/ViewSearchListener.php]: Can we make this conditional if a type isn't already set?

Comment (ADmad) [inline src/Listener/ViewSearchListener.php]: I moved the default setting up before options from filter are merged.

File: docs/_partials/pages/index/filters.rst [hunk 4]
```diff
@@ -1,2 +1,38 @@
...
+
+                // Whether to use selectize for select fields. Default `true`.
+                'selectize' => true,
+
+                // The search behavior collection to use. Default "default".
+                'collection' => 'default',
+
+                // Fields config for generation filter inputs. If `null` the field
+                // inputs will be derived based on filter collection. Default `null``.
+                'fields' => null
```
Comment (josegonzalez) [inline docs/_partials/pages/index/filters.rst]: Can you show another example of what the value of `fields` should be?

File: docs/_partials/pages/index/filters.rst [hunk 5]
```diff
@@ -1,2 +1,68 @@
...
+            'field' => ['title', 'body'],
+            'form' => [
+                'class' => 'foobar'
+            ]
+        ])
+        ->add('category_id', 'Search.Value', [
+            'form' => [
+                'type' => 'select',
+                'autocomplete' => true,
+                'selectize' => true,
```
Comment (josegonzalez) [inline docs/_partials/pages/index/filters.rst]: Should we unset the selectize, autocomplete, and anything else when outputting form elements?

**Comment by josegonzalez** [comment]
Other than my last comment, this seems mergeable.

**Comment by josegonzalez** [comment]
Is there a way to disable search filters?

**Comment by ADmad** [comment]
> Is there a way to disable search filters?

Disable the ViewSearch listener :)

File: docs/_partials/pages/index/filters.rst (hunk 5 above)
Comment (ADmad) [inline docs/_partials/pages/index/filters.rst]: Hmm actually the code doesn't deal with these option if set under `form`, my bad. Once has to set required classes. I'll don't want to go about updating the code, will fix the docs.

**Pull request merged**